### PR TITLE
Added function uimessage which creates UI Semantics message

### PR DESCRIPTION
**Pull request by mateuszwyszynski**
Maybe there is a way to do some kind of pattern matching like in Scala instead of using if and else?

Lacks the closing button "x" feature.

### CODE REVIEW

File: R/dsl.R [hunk 1]
```diff
@@ -103,3 +103,46 @@ generate_random_id <- function(prefix, id_length = 20) {
   random_id <- paste(sample(letters, id_length, replace = TRUE), collapse = "")
   paste0(prefix, "-", random_id)
 }
+
+#' Create Semantic UI Message
+#' 
+#' This creates a message using UI Semantics
```
Comment (filipstachura) [inline R/dsl.R]: `using Semantic UI`?

File: R/dsl.R [hunk 2]
```diff
@@ -103,3 +103,46 @@ generate_random_id <- function(prefix, id_length = 20) {
...
+#' @param icon If the message is of the type 'icon', specify the icon. Look at http://semantic-ui.com/elements/icon.html for all possibilities.
+#' 
+#' @export
+
+uimessage <- function(header, content, type = "", icon) {
+  if (grepl("icon", type)) {
+    if(missing(icon)) {
+      stop("Type 'icon' requires an icon!")
+    } else if(length(content) > 1) {
+      div(class = paste("ui message", type),
```
Comment (filipstachura) [inline R/dsl.R]: we use 2 spaces indentation

File: R/dsl.R [hunk 3]
```diff
@@ -103,3 +103,46 @@ generate_random_id <- function(prefix, id_length = 20) {
...
+                      content %>% lapply(tags$li))))
+    } else {
+      div(class = paste("ui message", type),
+          uiicon(icon),
+          div(class = "content",
+              div(class = "header", header),
+              content))
+    }
+  } else {
+    if (length(content) > 1) {
```
Comment (filipstachura) [inline R/dsl.R]: Shorter alternative:
```
if (length(content) > 1) {
  content <- tags$ul(class = "list", content %>% lapply(tags$li))
}      
div(class = paste("ui message", type),
  div(class = "header", header),
  content
)
```

Comment (filipstachura) [inline R/dsl.R]: Please try to fix the whole function using this approach (hint: do the same with icon)

File: R/dsl.R [hunk 4]
```diff
@@ -103,3 +103,48 @@ generate_random_id <- function(prefix, id_length = 20) {
...
+#'
+#' @param header Header of the message
+#' @param content Content of the message. If it is a vector, creates a list of vector's elements
+#' @param type Type of the message. Look at https://semantic-ui.com/collections/message.html for all possibilities.
+#' @param icon If the message is of the type 'icon', specify the icon. Look at http://semantic-ui.com/elements/icon.html for all possibilities.
+#' @param closable Determines whether the message should be closable. Default is FALSE - not closable
+#'
+#' @export
+
+closable_messages <- "$('.message .close')
```
Comment (filipstachura) [inline R/dsl.R]: This should be above the documentation. Make it UPPER_CASE to indicate it's a constant value.

**Comment by filipstachura** [comment]
Let's wait for a second review from @krystian8207

File: R/dsl.R [hunk 5]
```diff
@@ -103,3 +103,47 @@ generate_random_id <- function(prefix, id_length = 20) {
   random_id <- paste(sample(letters, id_length, replace = TRUE), collapse = "")
   paste0(prefix, "-", random_id)
 }
+
+CLOSABLE_MESSAGES <- "$('.message .close')
```
Comment (krystian8207) [inline R/dsl.R]: I would place this constant after uimessage definition. It would keep more consistent form of document.

**Pull request merged**